modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs/js/blogs.js

```diff
@@ -128,11 +131,9 @@ export default class Blogs {
 			'.cover-image-caption'
 		);
 
-		Liferay.on('coverImageDeleted', this._removeCaption, this);
-		Liferay.on(
-			['coverImageUploaded', 'coverImageSelected'],
-			this._showCaption,
-			this
+		this._imageSelectorCoverImageSubscription = State.subscribe(
+			imageSelectorCoverImageAtom,
+			(data) => this._updateCaption(data)
```

Comment:
Doing this "manually" here. If this were a React component could use the `useLiferayState` hook instead.